base/reflection.jl

```diff
@@ -542,14 +546,16 @@ code_llvm(f::ANY, types::ANY=Tuple) = code_llvm(STDOUT, f, types)
 code_llvm_raw(f::ANY, types::ANY=Tuple) = code_llvm(STDOUT, f, types, false)
 
 """
-    code_native([io], f, types)
+    code_native([io], f, types, [syntax])
 
 Prints the native assembly instructions generated for running the method matching the given
 generic function and type signature to `io` which defaults to `STDOUT`.
+Switch assembly syntax using `syntax` symbol parameter set to `:att` for AT&T syntax or `:intel` for Intel syntax. Output is AT&T syntax by default.
 """
-code_native(io::IO, f::ANY, types::ANY=Tuple) =
-    print(io, _dump_function(f, types, true, false, false, false))
-code_native(f::ANY, types::ANY=Tuple) = code_native(STDOUT, f, types)
+code_native(io::IO, f::ANY, types::ANY=Tuple, syntax::Symbol=:att) =
+    print(io, _dump_function(f, types, true, false, false, false, syntax))
+code_native(f::ANY, types::ANY=Tuple) = code_native(STDOUT, f, types, :att)
```

Comment:
I think you need to put something like
`code_native(f::ANY, types::ANY=Tuple, syntax::Symbol=:att) =  code_native(STDOUT, f, types, syntax)`